persistence/sql/transaction/transaction.go

```diff
@@ -0,0 +1,70 @@
+package transaction
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach-go/v2/crdb"
+	"github.com/gobuffalo/pop/v6"
+	"github.com/jmoiron/sqlx"
+	"github.com/ory/x/errorsx"
+)
+
+type transactionContextType string
+
+const TransactionContextKey transactionContextType = "transactionConnection"
+
+func Transaction(ctx context.Context, conn *pop.Connection, f func(ctx context.Context, c *pop.Connection) error) error {
```

Comment:
Could we use `popx` instead? It seems that the functions have indentical functionalities

Reply:
Done. The implementation is a bit hacky where we need to check whether popx/GetConnection returned the fallback; lmk if I should add something along the lines of `GetConnectionOrNil` to popx to make the check below cleaner:
```
	fallback := &pop.Connection{TX: &pop.Tx{}}
	tx := popx.GetConnection(ctx, fallback)
	if tx.TX == fallback.TX || tx.TX == nil {
```